Analysis/include/BaseEventAnalyzer.h

```diff
@@ -17,6 +18,16 @@ struct AnalyzerArguments : CoreAnalyzerArguments {
     OPT_ARG(std::string, output_sync,"sync.root");
 };
 
+struct SyncDescriptor {
+    std::shared_ptr<htt_sync::SyncTuple> sync_tree;
+    std::shared_ptr<boost::regex> regex_pattern;
+
+    SyncDescriptor(std::shared_ptr<htt_sync::SyncTuple> _sync_tree,
```

Comment:
move parse and syncTuple creation logic into the SyncDescriptor constructor. And move the constructor implementation into .cpp